src/surfaces/filtersDrawer/filtersDrawer.jsx

```diff
@@ -323,7 +329,10 @@ class FiltersDrawer extends React.Component {
                                             nestedTogglesOptionLabelKeyNum += 1;
                                             const isSelected = some(
                                                 nestedTogglesData.value,
-                                                option,
+                                                (v) => {
+                                                    const selectedValue = isObject(v) ? v.value : v;
+                                                    return option.value === selectedValue;
```

Comment:
Is there any way we can avoid having to "sniff" if we're a primitive or an option shaped object (`{ label, value }`) here, as well as down on Lines 598 and 644?  Was there a particular use case you ran into that really is different than other use cases using this component so far?  Is this what you mean by "allowing to use just `value` data for selection"?

I'm not necessarily opposed, but I want to ensure that this is the best way, before we complicate the code.

Reply:
@groberts314 you are right - at the moment you should "sniff" option shaped object for correct initialization, and I'm really sure that this a helpful update. I had the Dropdown similar functionality in mind.